code/edgemedia/src/test/java/com/adobe/marketing/mobile/edge/media/internal/MediaXDMEventGeneratorTests.kt

```diff
@@ -0,0 +1,792 @@
+        verify(mockEventProcessor, times(3)).processEvent(capture(sessionIdCaptor), capture(eventCaptor))
+        assertEquals(expectedEvent, eventCaptor.value)
+    }
+
+    @Test
+    fun testCustomMainPingIntervalAndDefaultAdPingInterval() {
+        // setup
+        val trackerConfig = mapOf<String, Any>(
+            MediaConstants.TrackerConfig.MAIN_PING_INTERVAL to 15
+        )
+        eventGenerator = MediaXDMEventGenerator(mediaContext, mockEventProcessor, trackerConfig, 0)
+
+        updateTs(15 * 1000)
+
+        val mediaCollection = XDMMediaCollection()
+        mediaCollection.playhead = getPlayhead()
+        var expectedEvent = XDMMediaEvent(XDMMediaSchema(XDMMediaEventType.PING, getDateFormattedTimestampFor(mockTimestamp), mediaCollection))
+
+        // test main content ping interval
+        eventGenerator.processPlayback()
+
+        // verify main content ping interval
+        verify(mockEventProcessor).processEvent(capture(sessionIdCaptor), capture(eventCaptor))
+        assertEquals(expectedEvent, eventCaptor.value)
+
+        // mock adStart
+        mediaContext.setAdInfo(AdInfo.create("id", "ad", 1, 15.0), mapOf())
+        updateTs(MediaInternalConstants.PingInterval.REALTIME_TRACKING_MS)
+
+        mediaCollection.playhead = getPlayhead()
+        expectedEvent = XDMMediaEvent(XDMMediaSchema(XDMMediaEventType.PING, getDateFormattedTimestampFor(mockTimestamp), mediaCollection))
+
+        // test ad content ping interval
+        eventGenerator.processPlayback()
+
+        // verify ad content ping interval
+        verify(mockEventProcessor, times(2)).processEvent(capture(sessionIdCaptor), capture(eventCaptor))
+        assertEquals(expectedEvent, eventCaptor.value)
+
+        // mock ad exit
+        mediaContext.clearAdInfo()
+        updateTs(15 * 1000)
+        eventGenerator.processPlayback()
+
+        mediaCollection.playhead = getPlayhead()
+        expectedEvent = XDMMediaEvent(XDMMediaSchema(XDMMediaEventType.PING, getDateFormattedTimestampFor(mockTimestamp), mediaCollection))
+
+        // verify main content ping interval after exiting from ad
+        verify(mockEventProcessor, times(3)).processEvent(capture(sessionIdCaptor), capture(eventCaptor))
+        assertEquals(expectedEvent, eventCaptor.value)
+    }
+
+    private fun updateTs(interval: Int, updatePlayhead: Boolean = true, reset: Boolean = false) {
+        if (reset) {
+            mockPlayhead = 0
+            mockTimestamp = 0
+        }
+        mockTimestamp += interval
+        if (updatePlayhead) {
+            mockPlayhead += (interval / 1000)
+            mediaContext.playhead = mockPlayhead.toDouble()
+        }
+
+        mediaContext.playhead = mockPlayhead.toDouble()
+        eventGenerator.setRefTS(mockTimestamp)
+    }
+
+    private fun getPlayhead(): Long {
+        return mediaContext.playhead.toLong()
+    }
+    private fun getDateFormattedTimestampFor(value: Long): Date {
+        return Date(value)
+    }
```

Comment:
I feel this function adds confusion as to me it implies there is some extra formatting going on for the test cases. Instead can you just use `Date(value)` and remove this function?